src/pytest_qgis/pytest_qgis.py

```diff
@@ -150,6 +150,8 @@ def qgis_app(request: "SubRequest") -> QgsApplication:
 
     if not request.config._plugin_settings.qgis_init_disabled:
         assert _APP
+        if not sip.isdeleted(_CANVAS):
+            sip.delete(_CANVAS)
```

Comment:
Would 
```python
_CANVAS.deleteLater()
```
work here? I think it is recommended for QObjects.

Reply:
`deleteLater` doesn't work but `delete` does. I'll change this.